profiles/database.py

```diff
@@ -0,0 +1,27 @@
+from datetime import datetime, timezone
+from typing import Optional, Type
+
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy import types
+
+db = SQLAlchemy()
+
+
+class UTCDateTime(types.TypeDecorator):
+    impl = types.DateTime
+
+    def python_type(self) -> Type:
+        return datetime
+
+    def process_bind_param(self, value, dialect) -> Optional[datetime]:
+        if value is not None:
+            return value.astimezone(timezone.utc)
```

Comment:
if I understand correctly, this decorator will transform the timezone of a `datetime` object when it is stored into the database, and then assume UTC when it takes it out.
However, this could lead to different behavior where a `OrcidToken` object is assigned a `datetime` object with the wrong timezone, and is used before saving it in the same process? In that case it would show the original, wrong timezone?
This normalization could be done in the domain object instead, whenever the field is assigned (as it shouldn't be assigned without indirection from outside the object).

Reply:
That's true, will change on initialisation too.

Reply:
This type is needed, unless we have a getter that converts it to UTC.

Reply:
So the type should only convert on reading but does not need to intervene on writing?

Reply:
Correct. Think we can remove this method.

Reply:
Hmm, the linting gods demand it. `NotImplementedError` works.

Reply:
Or disable `abstract-method`. I guess tests would fail if things aren't implemented correctly.